verification_token/models.py

```diff
@@ -0,0 +1,132 @@
+from datetime import timedelta
+from secrets import token_urlsafe
+
+from django.contrib.auth import get_user_model
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+from palvelutarjotin import settings
+
+
+class VerificationTokenManager(models.Manager):
+    def deactivate_token(self, obj_or_class, key=None, verification_type=None):
+        """
+        Deactivate a token. Key parameters is used and must be given
+        when using a model class instead of model instance.
+        """
+        self.filter_active_tokens(obj_or_class, key, verification_type).update(
+            is_active=False
+        )
+
+    def deactivate_and_create_token(
+        self,
+        obj,
+        user,
+        verification_type,
+        expiry_days=None,
+        deactivate_all_types_of_token=False,
+    ):
+        """
+        Deactivate old tokens (of a type) and create a new one.
+        """
+        if deactivate_all_types_of_token:
+            self.deactivate_token(obj, verification_type=None)
+        else:
+            self.deactivate_token(obj, verification_type=verification_type)
+
+        return self.create_token(obj, user, verification_type, expiry_days)
```

Comment:
Deactivates all old tokens of a given type related to an model instance and also creates a new one. If using deactivate_all_types_of_token -argument, all type of tokens can be deactivated at once.